superset-frontend/plugins/plugin-chart-echarts/src/utils/series.ts

```diff
@@ -52,20 +57,24 @@ export function extractDataTotalValues(
     stack: StackType;
     percentageThreshold: number;
     xAxisCol: string;
+    legendState?: LegendState;
   },
 ): {
   totalStackedValues: number[];
   thresholdValues: number[];
 } {
   const totalStackedValues: number[] = [];
   const thresholdValues: number[] = [];
-  const { stack, percentageThreshold, xAxisCol } = opts;
+  const { stack, percentageThreshold, xAxisCol, legendState } = opts;
   if (stack) {
     data.forEach(datum => {
       const values = Object.keys(datum).reduce((prev, curr) => {
         if (curr === xAxisCol) {
           return prev;
         }
+        if (legendState && !legendState[curr]) {
```

Comment:
nit: you can shorten this like

```suggestion
        if (!legendState?.[curr]) {
```

Reply:
If there's no `legendState` then it should skip this statement. It should only enter if `legendState` is defined and the value is not there.